apps/explorer/src/pages/transaction-result/transaction-summary/ObjectChanges.tsx

```diff
@@ -0,0 +1,335 @@
+                        title="Changes"
+                        size="sm"
+                        shadow="default"
+                        footer={
+                            renderFooter && (
+                                <div className="flex items-center justify-between">
+                                    <Text
+                                        variant="pBody/medium"
+                                        color="steel-dark"
+                                    >
+                                        {type === 'Shared' ? 'Shared' : 'Owner'}
+                                    </Text>
+                                    {type === 'AddressOwner' && (
+                                        <AddressLink address={locationId} />
+                                    )}
+                                    {type === 'ObjectOwner' && (
+                                        <ObjectLink objectId={locationId} />
+                                    )}
+                                    {type === 'Shared' && (
+                                        <Text
+                                            variant="pBody/medium"
+                                            color="hero-dark"
+                                        >
+                                            {locationId}
+                                        </Text>
+                                    )}
+                                </div>
+                            )
+                        }
+                    >
+                        <TransactionCardSection
+                            expanded={expanded}
+                            setExpanded={setExpanded}
+                            title={
+                                <Text
+                                    variant="body/semibold"
+                                    color="success-dark"
+                                >
+                                    {title}
+                                </Text>
+                            }
+                        >
+                            {changes.map(({ objectId, objectType }) => (
+                                <ObjectDetail
+                                    key={objectId}
+                                    objectId={objectId}
+                                    objectType={objectType}
+                                />
+                            ))}
+                        </TransactionCardSection>
+                    </TransactionCard>
+                );
+            })}
+        </>
+    );
+}
+
+function groupByOwner(objectSummaryChanges: any) {
+    if (!objectSummaryChanges) {
+        return {};
+    }
+
+    return objectSummaryChanges?.reduce(
+        (mapByOwner: Record<string, any>, change: any) => {
+            const owner = change?.owner;
+
+            let key = '';
+            if ('AddressOwner' in owner) {
+                key = `AddressOwner${OWNER_GROUP_DIVIDER}${owner.AddressOwner}`;
```

Comment:
You probably don't need any of this special logic, just use the object ID or the address owner.

Reply:
If you do have to do this why not just have two lists? Or add this information to the value so that you don't need to do this weird key compute

Reply:
I can do multiple lists here, but will there be an event where the same ID appears in both `AddressOwner` and `ObjectOwner` ?